app/_hub/kong-inc/vault-auth/index.md

```diff
@@ -33,39 +32,46 @@ params:
     - name: access_token_name
       required: true
       default: "`access_token`"
+      datatype: array of string elements
       description: |
-        Describes an array of comma-separated parameter names where the plugin will look for an access token. The client must send the access token in one of those key names, and the plugin will try to read the credential from a header or the querystring parameter with the same name.<br>*note*: the key names may only contain [a-z], [A-Z], [0-9], [_] and [-].
+        Describes an array of comma-separated parameter names where the plugin will look for an access token. The client must send the access token in one of those key names, and the plugin will try to read the credential from a header or the querystring parameter with the same name. The key names can only contain [a-z], [A-Z], [0-9], [_], and [-].
     - name: vault.id
       required: true
       default:
       value_in_examples: "<UUID>"
+      datatype: foreign ID
       description: |
         A reference to an existing `vault` object within the database. `vault` entities define the connection and authentication parameters used to connect to a Vault HTTP(S) API.
     - name: secret_token_name
       required: true
       default: "`secret_token`"
+      datatype: array of string elements
       description: |
-        Describes an array of comma-separated parameter names where the plugin will look for a secret token. The client must send the secret in one of those key names, and the plugin will try to read the credential from a header or the querystring parameter with the same name.<br>*note*: the key names may only contain [a-z], [A-Z], [0-9], [_] and [-].
-    - name: key_in_body
-      required: false
+        Describes an array of comma-separated parameter names where the plugin will look for a secret token. The client must send the secret in one of those key names, and the plugin will try to read the credential from a header or the querystring parameter with the same name. The key names can only contain [a-z], [A-Z], [0-9], [_], and [-].
+    - name: tokens_in_body
+      required: true
       default: "`false`"
+      datatype: boolean
       description: |
         If enabled, the plugin will read the request body (if said request has one and its MIME type is supported) and try to find the key in it. Supported MIME types are `application/www-form-urlencoded`, `application/json`, and `multipart/form-data`.
     - name: hide_credentials
-      required: false
+      required: true
       default: "`false`"
+      datatype: boolean
       description: |
         An optional boolean value telling the plugin to show or hide the credential from the upstream service. If `true`, the plugin will strip the credential from the request (i.e. the header or querystring containing the key) before proxying it.
     - name: anonymous
       required: false
       default:
+      datatype: string UUID
```

Comment:
I've seen UUID and i've seen string as a datatype in the past, generally not both. A UUID is a string, but it's a very specific one - a subset of the string datatype. Honestly, changing this to simply "UUID" will be just as clear, if not clearer, than the current label.

Reply:
> I've seen UUID and i've seen string as a datatype in the past, generally not both. A UUID is a string, but it's a very specific one - a subset of the string datatype. Honestly, changing this to simply "UUID" will be just as clear, if not clearer, than the current label.

@lena-larionova and @murillopaula the schema entry for anonymous is:

 { anonymous = { type = "string", uuid = true, legacy = true }, },

The description mentions that this is a consumer uuid so I think this should just be string.